File: 0722-remove-comments/0722-remove-comments.go
import "strings"

func removeComments(source []string) []string {
    isMultiline := false
    isSingleline := false
    rslt := ""
    
    joinedSource := strings.Join(source, "\\n")
    for i := 0; i < len(joinedSource); i++ {
        if i + 1 < len(joinedSource) && string(joinedSource[i]) == "/" && string(joinedSource[i+1]) == "*" && !isSingleline && !isMultiline{
            isMultiline = true
            i+=1
            continue
        }

        if i + 1 < len(joinedSource) && string(joinedSource[i]) == "*" && string(joinedSource[i+1]) == "/" && isMultiline{
            isMultiline = false
            i+=1
            continue
        }
        
        if isMultiline{
            continue
        }

        if i + 1 < len(joinedSource) && string(joinedSource[i]) == "/" && string(joinedSource[i+1]) == "/"{
            isSingleline = true
            i+=1
            continue
        }

        if i + 1 < len(joinedSource) && string(joinedSource[i]) == "\\" && string(joinedSource[i+1]) == "n"{
            isSingleline = false
            i+=1
            rslt += "\n"
            continue
        }

        if isSingleline {
            continue
        }

        rslt += string(joinedSource[i])
    }

 
    // fmt.Println("rslt", rslt)
    // fmt.Println("splitted", strings.Split(rslt,"\n"))
    splitted := strings.Split(rslt,"\n")
    no_empty := []string{}

    for i:=0; i<len(splitted); i++ {
        if splitted[i] != "" {
            no_empty = append(no_empty, splitted[i])
        }
    }

    return no_empty
}

File: 0722-remove-comments/0722-remove-comments.py
class Solution:
    def removeComments(self, source: List[str]) -> List[str]:
        isMultiline = False
        isOneline = False

        source_str = "\\n".join(source)
        print(source_str)

        i = 0
        rslt = ""
        while i < len(source_str):
            print(i)
            if i+1 < len(source_str):
                # print(source_str[i:i+2])
                if source_str[i:i+2] == "/*" and not isOneline and not isMultiline:
                    isMultiline = True
                    i+=2
                    continue
                
                elif source_str[i:i+2] == "*/" and isMultiline:
                    isMultiline = False
                    i+=2
                    continue
                
                elif source_str[i:i+2] == "//" and not isMultiline:
                    isOneline = True
                    i+=2
                    continue
                
                elif source_str[i:i+2] == "\\n" and isOneline:
                    rslt+="\\n"
                    isOneline = False
                    i+=2
                    continue
            
            if isMultiline:
                i+=1
                continue
            
            if isOneline:
                i+=1
                continue
            
            rslt += source_str[i]
            i+=1
        
        print("rslt", rslt)
        splitted_rslt = rslt.split("\\n")
        splitted_rslt = [x for x in splitted_rslt if len(x) > 0 ]
        return splitted_rslt